tools/setup-cmake.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
```

Comment:
This works across all distros to install more recent CMake, if build system needs 3.18, for example. Technically we can decide on what's min required whenever we need to ask for a new CMake feature, like `ninja` support - we can make sure our build scripts deploy the needed CMake version first before triggering the build.